milestones/exceptions.py

```diff
@@ -0,0 +1,45 @@
+"""
+Application-specific exception classes used throughout the implementation
+"""
+
+
+class InvalidCourseKeyException(Exception):
+    """
+    CourseKey validation exception class
+    """
+    pass
+
+
+class InvalidContentKeyException(Exception):
+    """
+    Course content/module/usage key validation exception class
+    """
+    pass
+
+
+class InvalidMilestoneException(Exception):
+    """
+    Milestone validation exception class
+    """
+    pass
+
+
+class InvalidMilestoneRelationshipTypeException(Exception):
+    """
+    Milestone Relationship Type validation exception class
+    """
+    pass
+
+
+class InvalidUserException(Exception):
+    """
+    User validation exception class
+    """
+    pass
+
+
+def raise_exception(entity_type, entity, exception):
+    """ Exception helper """
+    raise exception(
+        'The {} you have provided is not valid: {}'.format(entity_type, entity)
+    )
```

Comment:
nitpick: Seems like this is a ValidationError which has a Django defined class. You might want to consider using that class.